presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -471,7 +481,15 @@ private boolean directory(Path path)
     private boolean deleteObject(String key)
     {
         try {
-            s3.deleteObject(getBucketName(uri), key);
+            DeleteObjectRequest deleteObjectRequest = new DeleteObjectRequest(getBucketName(uri), key);
+            if (requesterPaysEnabled) {
+                // Not using request.setRequesterPays(true).
+                // Check https://github.com/aws/aws-sdk-java/issues/1219 for more details.
+                // TODO: When the issue is fixed, we can replace putCustomRequestHeader with withRequesterPays
+                deleteObjectRequest.putCustomRequestHeader(Headers.REQUESTER_PAYS_HEADER, Constants.REQUESTER_PAYS);
```

Comment:
My IDE pulls aws-java-sdk-s3-1.11.602 and shows `com.amazonaws.services.s3.model.DeleteObjectRequest#withRequesterPays` method. Could it be that they added the method even though the referenced issue is not closed yet?

Reply:
Yes, the method is withRequesterPays is available but the SDK code still doesn't set the relevant headers when it is set to true.

Reply:
@sakshia125 that's ... wow.
Let's rewrite the comment
```
// TODO use deleteObjectRequest.setRequesterPays() when https://github.com/aws/aws-sdk-java/issues/1219 is fixed
// currently the method exists, but is ineffective (doesn't set the required HTTP header)
```